specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-10-17-preview/workbookTemplates_API.json

```diff
@@ -0,0 +1,467 @@
+            "type": "string"
+          },
+          "description": "Resource tags"
+        }
+      },
+      "required": [
+        "location"
+      ],
+      "x-ms-azure-resource": true,
+      "description": "An azure resource object"
+    },
+    "WorkbookTemplatesListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/WorkbookTemplate"
+          },
+          "description": "An array of workbook templates."
+        }
+      },
+      "description": "WorkbookTemplate list result."
+    },
+    "WorkbookTemplate": {
+      "description": "An Application Insights workbook template definition.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/WorkbookTemplateResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Metadata describing a workbook template for an Azure resource.",
+          "$ref": "#/definitions/WorkbookTemplateProperties"
+        }
+      }
+    },
+    "WorkbookTemplateProperties": {
+      "description": "Properties that contain a workbook template.",
+      "required": [
+        "id",
+        "templateData",
+        "galleries"
+      ],
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Id of the workbook template. This is GUID value."
+        },
+        "priority": {
+          "type": "integer",
+          "description": "Priority of the template. Determines which template to open when a workbook gallery is opened in viewer mode."
+        },
+        "author": {
+          "type": "string",
+          "description": "Information about the author of the workbook template."
+        },
+        "templateData": {
+          "type": "object",
+          "description": "Valid JSON object containing workbook template payload."
+        },
+        "galleries": {
+          "type": "array",
+          "description": "Workbook galleries supported by the template.",
+          "items": {
+            "$ref": "#/definitions/WorkbookTemplateGallery"
+          }
+        },
+        "localized": {
+          "type": "object",
+          "description": "Key value pair of localized gallery. Each key is the locale code of languages supported by the Azure portal.",
+          "additionalProperties": {
+            "type": "array",
+            "items": {
+              "$ref": "#/definitions/WorkbookTemplateLocalizedGallery"
+            }
+          }
+        }
+      }
+    },
+    "WorkbookTemplateGallery": {
+      "description": "Gallery information for a workbook template.",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of the workbook template in the gallery."
+        },
+        "category": {
+          "type": "string",
+          "description": "Category for the gallery."
+        },
+        "type": {
```

Comment:
Can this be a string based enum?  How many allowed values are there?

Reply:
This is going to be an ever growing list. As we support more and more scenarios that show workbooks based insights, this field is going to have many values.